hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/PartitionStatsIndexSupport.scala

```diff
@@ -133,6 +137,10 @@ class PartitionStatsIndexSupport(spark: SparkSession,
     }
   }
 
+  def canLookupInPSI(queryFilters: Seq[Expression], indexedCols: Seq[String]): Boolean = {
+    queryFilters.exists(DataSkippingUtils.containsNullOrValueCountBasedFilters(_, indexedCols = indexedCols))
```

Comment:
Shouldn't it be `!queryFilters.exists(...)` i.e. look up in PSI only if the filters **do not** contain any null or value-count references?

Reply:
fixed the logic and added a test to validate PSI skipped for a null filter on an indexed column